File: app/src/main/kotlin/metamark/parser/MatchZeroOrOneExpressionTerm.kt
package metamark.parser

import metamark.parser.antlr.MetamarkParser

data class MatchZeroOrOneExpressionTerm(
    val optionalExpression: ExpressionTerm,
): ExpressionTerm() {
    companion object {
        fun build(
            ctx: MetamarkParser.Expression_matchZeroOrOneExpressionContext,
        ): MatchZeroOrOneExpressionTerm {
            return MatchZeroOrOneExpressionTerm(
                optionalExpression = ExpressionTerm.build(ctx.optionalExpression)
            )
        }
    }
}

File: src/main/MatchZeroOrOneExpressionTerm.ts
import {ExpressionTerm} from './ExpressionTerm';
import {Expression_matchZeroOrOneExpressionContext} from "../generated_parser/MetamarkParser";

export class MatchZeroOrOneExpressionTerm extends ExpressionTerm {
    static build(ctx: Expression_matchZeroOrOneExpressionContext): MatchZeroOrOneExpressionTerm {
        return new MatchZeroOrOneExpressionTerm(ExpressionTerm.build(ctx._optionalExpression));
    }

    optionalExpression: ExpressionTerm;

    constructor(optionalExpression: ExpressionTerm) {
        super();
        this.optionalExpression = optionalExpression;
    }
}
